OpenRA.Game/Renderer.cs

```diff
@@ -379,6 +387,19 @@ public IVertexBuffer<Vertex> CreateVertexBuffer(int length)
 			return Context.CreateVertexBuffer(length);
 		}
 
+		public IIndexBuffer CreateQuadIndexBuffer(int length)
```

Comment:
This would be better placed in `OpenRA.Game/Graphics/Util.cs`.

Reply:
Renderer contains a QuadIndexBuffer, so I don't think that the current place is wrong

Reply:
The issue is (still) about the pre-population of the index buffer assuming a specific vertex layout. Renderer doesn't know or care about this because it uses higher level objects that eventually call into the `FastCreateQuad` methods defined in `OpenRA.Game/Graphics/Util.cs`. Those methods are tied to the index buffer format, so that needs to be specified in the same file.

Looking at the latest implementation in more detail, I will go further to suggest that `Context.CreateIndexBuffer()` should take the buffer length as an argument to keep symmetry with the vertex buffer, the `StaticIndexBuffer` should define `SetData` methods to set the indices (again for symmetry with VertexBuffer), and we have a `Util.FillQuadIndexBuffer` alongside the `FastCreateQuad`s

Reply:
As we don't need the index buffer to be dynamic I specifically pass in a value to OpenGL that promises that we aren't going to be changing it. SetData would be violating that optimisation.

Even for voxel renderer on change we are likely going to be disposing of the old index buffer and creating a new static buffer. Mirroring its current vertex buffer implementation

Reply:
It would be useful to see some numbers about whether that optimisation has any real world impact (I suspect it won't). In any case, the rest of my comment still stands.

Reply:
moved to utils. I hope it's the end of bikeshedding